src/queueloader.cpp

```diff
@@ -206,6 +215,22 @@ void QueueLoader::reload(std::vector<Download>& downloads, bool remove_unplayed)
 		f.close();
 	}
 
+	if (ctrl->get_cfgcont()->get_configvalue_as_bool("delete-downloaded-files")) {
+		for (const auto& dl : deletion_list) {
+			std::string filename = dl.filename();
+			if (access(filename.c_str(), F_OK) == 0) {
```

Comment:
This creates a [TOCTOU bug](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use), but luckily this check can be avoided: we can just run `std::remove`, check its exit code, and inspect `errno`, logging errors only if `errno != ENOENT`.

Reply:
Oops, fixed